meteor/sofie-server/data-access/repositories/mongo/mongo-ad-lib-piece-repository.ts

```diff
@@ -0,0 +1,38 @@
+import { BaseMongoRepository } from './base-mongo-repository'
+import { AdLibPieceRepository } from '../interfaces/ad-lib-piece-repository'
+import { Identifier } from '../../../model/interfaces/identifier'
+import { MongoAdLibPiece, MongoEntityConverter } from './mongo-entity-converter'
+import { MongoDatabase } from './mongo-database'
+import { AdLibPiece } from '../../../model/entities/ad-lib-piece'
+import { NotFoundException } from '../../../model/exceptions/not-found-exception'
+
+const AD_LIB_COLLECTION_NAME: string = 'adLibPieces'
+
+export class MongoAdLibPieceRepository extends BaseMongoRepository implements AdLibPieceRepository {
+	constructor(mongoDatabase: MongoDatabase, mongoEntityConverter: MongoEntityConverter) {
+		super(mongoDatabase, mongoEntityConverter)
+	}
+
+	protected getCollectionName(): string {
+		return AD_LIB_COLLECTION_NAME
+	}
+
+	public async getAdLibPieceIdentifiers(rundownId: string): Promise<Identifier[]> {
+		this.assertDatabaseConnection('getAdLibIdentifiers')
```

Comment:
Should the string argument match the name of the method like in the method below?

If so then doing the following, should also help reducing the changes needed in the future, should a method be renamed.

`this.assertDatabaseConnection(this.getAdLibPieceIdentifiers.name)`

Reply:
This has been implemented :-)